repos/system_upgrade/el7toel8/libraries/rhsm.py

```diff
@@ -93,57 +95,121 @@ def _no_op(*args, **kwargs):
+            'The `yum repoinfo` command reports that the following repositories are defined multiple times:\n{0}'
+            .format('\n'.join(duplicates))
+        ),
+        reporting.Severity(reporting.Severity.MEDIUM),
+        reporting.Tags([reporting.Tags.REPOSITORY]),
+        reporting.Flags([reporting.Flags.INHIBITOR]),
+        reporting.Remediation(hint='Remove the duplicit repository definitions.')
+    ])
+
+
+def _get_repos(repos_raw_stdout):
+    """
+    Generator providing all the repos available through yum/dnf.
+
+    :rtype: Iterator[:py:class:`leapp.libraries.common.rhsm.Repo`]
+    """
+    # Split all the available repos per one repo
+    for repo_raw_params in re.findall(
+            r"Repo-id.*?Repo-filename.*?\n", repos_raw_stdout, re.DOTALL | re.MULTILINE):
+        yield _parse_repo_params(repo_raw_params)
+
+
+def _parse_repo_params(repo_raw_params):
+    """Parse multiline string holding repo parameters to distill the important ones."""
+    try:
+        repoid = _get_repo_param(r'^Repo-id\s+:\s+([^/]+).*?$', repo_raw_params, 'Repo-id')
+        repofile = _get_repo_param(r'^Repo-filename:\s+(.*?)$', repo_raw_params, 'Repo-filename')
+        return namedtuple('Repository', ['repoid', 'file'])(repoid, repofile)
+    except ValueError, err:
+        api.current_logger().warn("Failed to parse the '%s' repo parameter of the `yum repoinfo` output", err.args[0])
+
+
+def _get_repo_param(pattern, repo_raw_params, param):
+    """Parse a string with all the repo params to get the value of a single repo param."""
+    sub_attr = re.search(pattern,
+                         repo_raw_params,
+                         re.MULTILINE | re.DOTALL)
+    if sub_attr:
+        return sub_attr.group(1)
+    else:
+        raise ValueError(param, repo_raw_params)
 
 
 @with_rhsm
-def get_enabled_repo_uids(context, rhsm_info):
+def get_enabled_repo_ids(context):
     """
-    Retrieves all enabled repositories names from the subscription manager.
+    Retrieve repo ids of all the repositories enabled through the subscription-manager.
 
     :param context: An instance of a mounting.IsolatedActions class
     :type context: mounting.IsolatedActions class
-    :param rhsm_info: An instance of a RHSMInfo derived model.
-    :type rhsm_info: RHSMInfo derived model
+    :return: Repositories that are enabled on the current system through the subscription-manager.
+    :rtype: List(string)
     """
-    if not rhsm_info.enabled_repos:
-        with _handle_rhsm_exceptions():
-            result = context.call(['subscription-manager', 'repos', '--list-enabled'], split=False)
-            rhsm_info.enabled_repos = _RE_REPO_UID.findall(result['stdout'])
+    with _handle_rhsm_exceptions():
+        result = context.call(['subscription-manager', 'repos', '--list-enabled'], split=False)
+        return _RE_REPO_UID.findall(result['stdout'])
 
 
 @with_rhsm
 def _get_repositories_to_use(context, rhsm_info, target_repositories):
```

Comment:
@vinzenz, @pirat89, this function is not used anywhere. And it seems it has not been ever. Do you know something about it? I would just remove it.

Reply:
As discussed, I suggest to remove it in separate commit (in this PR). It seems that it's really death-code.

Reply:
Oh, I hope it does not bring death to anyone. Let's get rid of it before it does: https://github.com/oamg/leapp-repository/pull/468